extension/demo/src/bg/background.js

```diff
@@ -3,64 +3,80 @@
 import browser from 'webextension-polyfill'
 
 import { addListener } from '../modules/messaging'
-import { API_BASE, STORAGE_KEY } from '../modules/constants'
-
-async function getIsAuthenticated () {
-  const results = await browser.storage.local.get([STORAGE_KEY])
-  return !!results[STORAGE_KEY]
-}
-
-/**
- * @param {object} data Data to save to storage
- * @returns {Promise}
- */
-function updateStorage (data) {
-  const payload = {
-    [STORAGE_KEY]: {
-      auth: data
-    }
-  }
-  return browser.storage.local.set(payload)
-}
+import { ACCESS_TOKEN, API_BASE_WEB, PAYLOAD, TTL } from '../modules/constants'
+import * as api from '../modules/api'
+import * as storage from '../modules/storage'
 
 async function onPopupTriggerAuth () {
-  const isAuthenticated = await getIsAuthenticated()
-  if (!isAuthenticated) {
+  const accessToken = await storage.get(ACCESS_TOKEN)
+  if (!accessToken) {
     browser.tabs.create({
-      url: `${API_BASE}/oauth/ext`,
+      url: `${API_BASE_WEB}/oauth/ext`,
       active: true
     })
   }
 }
 
 function onPopupOpenApps () {
   browser.tabs.create({
-    url: `${API_BASE}/apps`,
+    url: `${API_BASE_WEB}/apps`,
     active: true
   })
 }
 
 async function onToken (data) {
-  await updateStorage(data)
-  const isAuthenticated = !browser.runtime.lastError
-  return {
-    isAuthenticated
+  const { accessToken } = data
+  const isAuthenticated = await storage.set(data)
+  updatePayload(accessToken)
+
+  return { isAuthenticated }
+}
+
+async function requestPayload () {
+  let response = {}
+  response[PAYLOAD] = await storage.get(PAYLOAD)
+
+  return response
+}
+
+function timeToLive (expiresAt) {
+  return expiresAt - (Math.floor(Date.now() / 1000) + 3600)
+}
+
+async function updatePayload (accessToken) {
+  const payload = await storage.get(PAYLOAD)
+  if (!payload) {
+    const subscriptionStatus = await api.fetchSubscriptionStatus(accessToken)
+    if (subscriptionStatus.hasOwnProperty(PAYLOAD)) {
+      storage.set(subscriptionStatus)
+    }
+  } else {
+    const ttl = await storage.get(TTL)
+    if (ttl) {
+      const callback = () => updatePayload(accessToken)
+      setTimeout(callback, timeToLive(ttl))
+    }
   }
 }
 
-function onSubscription (data) {
-  updateStorage(data)
+async function onSubscription (data) {
+  storage.set(data)
 }
```

Comment:
```suggestion
function onSubscription (data) {
  storage.set(data)
}
```

The function is defined as async but we're not waiting on anything.